rskj-core/src/main/java/co/rsk/remasc/Remasc.java

```diff
@@ -117,23 +140,24 @@ void processMinersFees() throws IOException, BlockStoreException {
         }
 
         // Takes from rewardBalance this block's height reward.
-        Coin fullBlockReward = rewardBalance.divide(BigInteger.valueOf(remascConstants.getSyntheticSpan()));
-        rewardBalance = rewardBalance.subtract(fullBlockReward);
+        Coin syntheticReward = rewardBalance.divide(BigInteger.valueOf(remascConstants.getSyntheticSpan()));
+        rewardBalance = rewardBalance.subtract(syntheticReward);
         provider.setRewardBalance(rewardBalance);
 
         // Pay RSK labs cut
-        Coin payToRskLabs = fullBlockReward.divide(BigInteger.valueOf(remascConstants.getRskLabsDivisor()));
+        Coin payToRskLabs = syntheticReward.divide(BigInteger.valueOf(remascConstants.getRskLabsDivisor()));
         feesPayer.payMiningFees(processingBlockHeader.getHash().getBytes(), payToRskLabs, remascConstants.getRskLabsAddress(), logs);
-        fullBlockReward = fullBlockReward.subtract(payToRskLabs);
+        syntheticReward = syntheticReward.subtract(payToRskLabs);
 
         RemascFederationProvider federationProvider = new RemascFederationProvider(config, repository, processingBlock);
 
-        Coin payToFederation = fullBlockReward.divide(BigInteger.valueOf(remascConstants.getFederationDivisor()));
+        Coin payToFederation = syntheticReward.divide(BigInteger.valueOf(remascConstants.getFederationDivisor()));
 
         byte[] processingBlockHash = processingBlockHeader.getHash().getBytes();
         int nfederators = federationProvider.getFederationSize();
-        Coin payToFederator = payToFederation.divide(BigInteger.valueOf(nfederators));
-        Coin restToLastFederator = payToFederation.subtract(payToFederator.multiply(BigInteger.valueOf(nfederators)));
+        Coin[] payAndRemainderToFederator = payToFederation.divideAndRemainder(BigInteger.valueOf(nfederators));
+        Coin payToFederator = payAndRemainderToFederator[0];
```

Comment:
Are we sure that we are going to always have index `0` and `1` after divideAndRemainder? What happens on nfederators = 1?